packages/flutterfire_cli/lib/src/commands/config.dart

```diff
@@ -557,17 +867,185 @@ class ConfigCommand extends FlutterFireCommand {
       final xcodeProjFilePath =
           path.join(flutterApp!.macosDirectory.path, 'Runner.xcodeproj');
 
-      final rubyScript =
-          generateRubyScript(googleServiceInfoFile, xcodeProjFilePath);
-
       if (Platform.isMacOS) {
-        final result = await Process.run('ruby', [
-          '-e',
-          rubyScript,
-        ]);
-
-        if (result.exitCode != 0) {
-          throw Exception(result.stderr);
+        // We need to prompt user whether they want a scheme configured, target configured or to simply write to the path provided
+        if (fullMACOSServicePath != null) {
+          final fileName = path.basename(fullMACOSServicePath!);
+          final response = promptSelect(
+            'Would you like your macOS $fileName to be associated with your macOS Scheme or Target (use arrow keys & space to select)?',
+            [
+              'Scheme',
```

Comment:
Should we move part of the functions to another file and separate it into named functions? Getting to thousands of lines is probably going to make the function hard to debug